worlds/oot_soh/location_access/overworld/market.py

```diff
@@ -5,8 +5,273 @@
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT11, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE1, lambda bundle: (is_child(bundle) and can_break_crates(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE2, lambda bundle: (is_child(bundle) and can_break_crates(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE3, lambda bundle: (is_child(bundle) and can_break_crates(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE4, lambda bundle: (is_child(bundle) and can_break_crates(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE5, lambda bundle: (is_child(bundle) and can_break_crates(bundle)))
+    ])
+    # Connections
+    connect_regions(Regions.MARKET_GUARD_HOUSE, world, [
+        (Regions.MARKET_ENTRANCE, lambda bundle: True)
+    ])
+    
+    ## Market Bazaar
+    # Locations
+    add_locations(Regions.MARKET_BAZAAR, world, [
+        (Locations.MARKET_BAZAAR_ITEM1, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM2, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM3, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM4, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM5, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM6, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM7, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM8, lambda bundle: True),
+    ])
+    # Connections
+    connect_regions(Regions.MARKET_BAZAAR, world, [
+        (Regions.MARKET, lambda bundle: True)
+    ])
+    
+    ## Market Mask Shop
+    # Events
+    add_events(Regions.MARKET_MASK_SHOP, world, [
+        (EventLocations.MARKET_MASK_SHOP_ALL_MASKS, Events.CAN_BORROW_MASKS, lambda bundle: (has_item(Items.ZELDAS_LETTER, bundle) and
+                                                                                   (has_item(Events.KAKARIKO_GATE_OPEN, bundle) or world.options.kakariko_gate.value == 1) or
+                                                                                   world.options.complete_mask_quest.value == 1)),
+        (EventLocations.MARKET_MASK_SHOP_SKULL_MASK, Events.CAN_BORROW_SKULL_MASK, lambda bundle: (world.options.complete_mask_quest.value or
+                                                                                        has_item(Events.CAN_BORROW_MASKS, bundle))),
+        (EventLocations.MARKET_MASK_SHOP_SPOOKY_MASK, Events.CAN_BORROW_SPOOKY_MASK, lambda bundle: (world.options.complete_mask_quest.value or
+                                                                                        has_item(Events.CAN_BORROW_MASKS, bundle))),
+        (EventLocations.MARKET_MASK_SHOP_BUNNY_HOOD, Events.CAN_BORROW_BUNNY_HOOD, lambda bundle: (world.options.complete_mask_quest.value or
+                                                                                        has_item(Events.CAN_BORROW_MASKS, bundle)))
```

Comment:
Could you replace the `world.options.complete_mask_quest.value` with `world.options.complete_mask_quest.value == 1`? I have a feeling I ran into issues last time trying to return an int where it essentially needs a bool.